verification/cbmc/proofs/aws_cryptosdk_priv_hdr_parse_alg_id/aws_cryptosdk_priv_hdr_parse_alg_id_harness.c

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/cryptosdk/header.h>
+#include <aws/cryptosdk/private/header.h>
+#include <make_common_data_structures.h>
+#include <proof_helpers/make_common_data_structures.h>
+#include <proof_helpers/proof_allocators.h>
+#include <proof_helpers/utils.h>
+
+void aws_cryptosdk_priv_hdr_parse_alg_id_harness() {
+    /* Nondet Input */
+    struct aws_cryptosdk_hdr *hdr                  = ensure_nondet_hdr_has_allocated_members(MAX_TABLE_SIZE);
+    struct aws_byte_cursor *pcursor                = malloc(sizeof(*pcursor));
+    struct aws_cryptosdk_alg_properties *alg_props = malloc(sizeof(*alg_props));
```

Comment:
Is it not necessary to assume that `alg_props` is non-NULL?

Reply:
No it seems like at least in terms of memory safety and the functional properties, we do check that alg_props can be NULL. I think because it's set in the function under test:
`*alg_props = aws_cryptosdk_alg_props(alg_id);`